### PR TITLE
Experimental DB2 support

### PR DESCRIPTION
**Pull request by eputtone**

### CODE REVIEW

File: nflow-engine/src/main/resources/scripts/db/db2.create.ddl.sql [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  executor_id int,
+  retries int not null default 0,
+  created timestamp(3) not null default current_timestamp,
+  modified timestamp(3) not null default current_timestamp,
+  executor_group varchar(64) not null,
+  workflow_signal int,
+  constraint nflow_workflow_uniq unique (type, external_id, executor_group)
+);
+
+-- create or replace function update_modified() returns trigger language plpgsql as '
```
Comment (jsyrjala) [inline nflow-engine/src/main/resources/scripts/db/db2.create.ddl.sql]: This commented stuff is Oracle related? Could be removed?

File: nflow-engine/src/main/java/io/nflow/engine/config/db/Db2DatabaseConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  @Override
+  protected void checkDatabaseConfiguration(Environment env, DataSource dataSource) {
+    JdbcTemplate jdbc = new JdbcTemplate(dataSource);
+    Long dbTimeZoneOffsetHours = jdbc.queryForObject("select current timezone from sysibm.sysdummy1", Long.class);
+    Long propsTimeZoneOffsetHours = HOURS.convert(
+            TimeZone.getTimeZone(property(env, "timezone")).getOffset(currentTimeMillis()), MILLISECONDS);
+    if (!Objects.equals(dbTimeZoneOffsetHours, propsTimeZoneOffsetHours)) {
+      throw new RuntimeException("Database has unexpected time zone - hour offset DB2: " + dbTimeZoneOffsetHours +
```
[the remark this answers is not in the document]
Comment (eputtone) [reply nflow-engine/src/main/java/io/nflow/engine/config/db/Db2DatabaseConfiguration.java]: Added

File: nflow-engine/src/main/resources/scripts/db/db2.create.ddl.sql [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  primary key (type, executor_group)
+);
+
+-- drop trigger if exists update_nflow_definition_modified on nflow_workflow_definition;
+-- create trigger update_nflow_definition_modified before update on nflow_workflow_definition for each row execute procedure update_modified();
+
+
+-- Archive tables
+-- - no default values
+-- - no default values
```
[the remark this answers is not in the document]
Comment (eputtone) [reply nflow-engine/src/main/resources/scripts/db/db2.create.ddl.sql]: Removed

File: nflow-engine/src/main/resources/scripts/db/db2.create.ddl.sql (hunk 1 above)
Comment (eputtone) [reply nflow-engine/src/main/resources/scripts/db/db2.create.ddl.sql]: It was just missing :). Integration tests didn't cover checking that modified-column auto-updates. Now they include at least one check.

File: nflow-engine/src/main/java/io/nflow/engine/config/db/Db2DatabaseConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.env.Environment;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import io.nflow.engine.internal.storage.db.SQLVariants;
+import io.nflow.engine.workflow.instance.WorkflowInstance.WorkflowInstanceStatus;
+
+/**
+ * Configuration for DB2 database. Note: tested only using DB2 Express-C (Docker: ibmcom/db2express-c).
+ * DB2 database must be configured to UTC time zone, otherwise nFlow will fail to function correctly.
```
Comment (gmokki) [inline nflow-engine/src/main/java/io/nflow/engine/config/db/Db2DatabaseConfiguration.java]: remove line about UTC

Comment (eputtone) [reply nflow-engine/src/main/java/io/nflow/engine/config/db/Db2DatabaseConfiguration.java]: Done

**Pull request merged**